docs/design/windows-design.md

```diff
@@ -210,8 +210,33 @@ Kube-proxy userspace mode is configured to provide NodePort Service function. A
 "HNS Internal NIC" is provided to kube-proxy to configure Service addresses. The OpenFlow entries for the
 NodePort Service traffic on Windows are the same as those on Linux.
 
-The ClusterIP Service function is implemented by Antrea leveraging OVS. Antrea installs OpenFlow entries
-to select the backend endpoint and performs DNAT on the traffic.
+AntreaProxy implements the ClusterIP Service function. Antrea Agent installs routes to send ClusterIP Service
+traffic from host network to the OVS bridge. For each Service, it adds a route that routes the traffic via a
+virtual IP (169.254.169.252), and it also adds a route to indicate that the virtual IP is reachable via
+antrea-gw0. The reason to add a virtual IP, rather than routing the traffic directly to antrea-gw0, is that
+then only one ARP responder flow is needed, which resolves the virtual IP to a virtual MAC.
+
+When a Service's endpoints are in hostNetwork, a request packet will have its destination IP DNAT'd to peer
+Node IP and its source IP will be SNAT'd to another virtual IP (169.254.169.253). Such SNAT is needed
+because for reply packets, the destination IP will be the antrea-gw0 IP if SNAT is not performed, then
+the packet will be routed to antrea-gw0 without entering OVS pipeline.
+
+For a request packet, it will enter OVS pipeline via antrea-gw0 and exit via antrea-gw0 as well to host network.
+On Windows host, with the help of NetNat, the request packet's source IP will be SNAT'd again to Node IP.
+
+The reply packets are the reverse for both situations when hostNetwork is false or true.
+
+A request packet with Service Endpoint in hostNetwork is:
+host -> antrea-gw0 -> OVS pipeline -> antrea-gw0 -> host NetNat -> br-int -> OVS pipeline -> peer Node
+                           |                            |
+                   DNAT(peer Node IP)              SNAT(Node IP)
+                    SNAT(virtual IP)
```

Comment:
Probably still add some words to explain why SNAT to virtual IP is necessary.

Reply:
Maybe I misunderstood something. I think here it is explained already.
https://github.com/antrea-io/antrea/pull/2235/files/f32339f05c92268403b3e7fdf674a6c1641187dc#diff-34b55715110627cfbedffde7ac38663b6c86987666f790ff431558404ae7eb5eR220

Reply:
Ok. I did not see that.